yt_dlp/extractor/ilpost.py

```diff
@@ -35,34 +37,75 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, display_id)
 
         endpoint_metadata = self._search_json(
-            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', display_id)
-        episode_id = endpoint_metadata['post_id']
-        podcast_id = endpoint_metadata['podcast_id']
-        podcast_metadata = self._download_json(
-            endpoint_metadata['ajax_url'], display_id, data=urlencode_postdata({
-                'action': 'checkpodcast',
-                'cookie': endpoint_metadata['cookie'],
-                'post_id': episode_id,
-                'podcast_id': podcast_id,
-            }))
+            r'{"props":{"pageProps":', webpage, 'metadata', display_id)
+        episode_id = endpoint_metadata['data']['data']['episode']['data'][0]['id']
+        podcast_id = traverse_obj(endpoint_metadata, ('data', 'data', 'episode', 'data', 0, 'parent', 'id'))
+        podcast_metadata = traverse_obj(endpoint_metadata, ('data', 'data', 'episode', 'data', 0))
```

Comment:
might as well just do this, and inline the `id`s below
```suggestion
        episode = endpoint_metadata['data']['data']['episode']['data'][0]
```